src/main/java/org/alliancegenome/curation_api/services/DiseaseAnnotationService.java

```diff
@@ -212,6 +213,7 @@ public void validateAnnotation(DiseaseAnnotation entity, boolean isUpdate) {
                 throw new ApiErrorException(response);
                 // do not continue validation for update if Disease Annotation ID has not been found
             }       
+            originalDoTerm = entity.getCurie();
```

Comment:
This is not the DoTerm this is the auto generated Curie. you might be looking for entity.getObject().getCurie()?